tests/testthat/test-mime.R

```diff
@@ -0,0 +1,148 @@
+  ) %>%
+    expect_equal("Sat, 15 Jun 2019 20:18:00 -0400")
+})
+
+test_that("varying formats for recipient lists work as expected", {
+
+  # Named vs. unnamed recipients (or some named and some unnamed)
+  #   Recipients with Unicode characters and double-quotes in their display names
+  # Single vs. multi recipients
+
+  # Create an empty email message
+  email <- compose_email()
+
+  email_date <- ISOdatetime(2015, 6, 2, 20, 18, 00, tz = "America/Los_Angeles")
+
+  # Test resulting MIME messages with variations
+  # in email address lists
+  generate_rfc2822(
+    eml = email,
+    date = email_date,
+    subject = "test",
+    from = "sender@example.com",
+    to = c("receive_1@example.com")
+  ) %>%
+    expect_match("To: <receive_1@example.com>")
+
+  generate_rfc2822(
+    eml = email,
+    date = email_date,
+    subject = "test",
+    from = "sender@example.com",
+    to = c("Receiver" = "receive_1@example.com")
+  ) %>%
+    expect_match("To: \"Receiver\" <receive_1@example.com>")
+
+  generate_rfc2822(
+    eml = email,
+    date = email_date,
+    subject = "test",
+    from = "sender@example.com",
+    to = c("receive_1@example.com", "receive_2@example.com")
+  ) %>%
+    expect_match("To: <receive_1@example.com>, <receive_2@example.com>")
+
+  generate_rfc2822(
+    eml = email,
+    date = email_date,
+    subject = "test",
+    from = "sender@example.com",
+    to = c("receive_1@example.com", "Receiver" = "receive_2@example.com")
+  ) %>%
+    expect_match("To: <receive_1@example.com>, \"Receiver\" <receive_2@example.com>")
+
+  generate_rfc2822(
+    eml = email,
+    date = email_date,
+    subject = "test",
+    from = "sender@example.com",
+    to = c("Receiv\U00E8r" = "receive_1@example.com")
+  ) %>%
+    expect_match("To: =\\?utf-8\\?B\\?UmVjZWl2w6hy\\?= <receive_1@example.com>")
+
+  generate_rfc2822(
+    eml = email,
+    date = email_date,
+    subject = "test",
+    from = "sender@example.com",
+    to = c("The \"Real\" Receiver" = "receive_1@example.com")
+  ) %>%
+    expect_match("To: \"The \\\\\"Real\\\\\" Receiver\" <receive_1@example.com>")
+
+  # Test the subject line with several variations
+  # (Unicode characters, double-quotes)
+  generate_rfc2822(
+    eml = email,
+    date = email_date,
+    subject = "Don't sw\U00E8at it",
+    from = "sender@example.com",
+    to = "receive_1@example.com"
+  ) %>%
+    expect_match("Subject: =\\?utf-8\\?B\\?RG9uJ3Qgc3fguop0IGl0\\?=")
+
+  generate_rfc2822(
+    eml = email,
+    date = email_date,
+    subject = "Don't \"sweat it\", okay?",
+    from = "sender@example.com",
+    to = "receive_1@example.com"
+  ) %>%
+    expect_match("Subject: Don't \"sweat it\", okay?")
+
+# TODO:
+# Inline images and file attachments with weird filenames
+#   Filenames with spaces
+#   Filenames with unicode characters (on both Windows and POSIX)
+#   Filenames with angle brackets < >
+#   With various extensions
```

Comment:
These scenarios are for manual testing but it'd be good to have an R script in the manualtests directory for these.